src/webgl/buffer.js

```diff
@@ -252,69 +224,75 @@ export default class Buffer extends Resource {
   }) {
     // TODO optimize dstData according to offset and length
     dstData = dstData || new ArrayBuffer(this.bytes);
+    this.gl.bindBuffer(GL.COPY_READ_BUFFER, this.handle);
     this.gl.getBufferSubData(this.target, srcByteOffset, dstData, dstOffset, length);
+    this.gl.bindBuffer(GL.COPY_READ_BUFFER, null);
     return dstData;
   }
 
   /**
    * Binds a buffer to a given binding point (target).
+   *   GL.TRANSFORM_FEEDBACK_BUFFER and GL.UNIFORM_BUFFER take an index, and optionally a range.
    *
    * @param {Glenum} target - target for the bind operation.
-   *  Possible values: gl.TRANSFORM_FEEDBACK_BUFFER and gl.UNIFORM_BUFFER
-   * @param {GLuint} index - the index of the target.
-   * @returns {Buffer} - Returns itself for chaining.
-   */
-  bind() {
-    this.gl.bindBuffer(this.target, this.handle);
-    return this;
-  }
-
-  unbind() {
-    this.gl.bindBuffer(this.target, null);
-    return this;
-  }
-
-  /**
-   * Note: WEBGL2
-   * Binds a buffer to a given binding point (target) at a given index.
    *
-   * @param {Glenum} target - target for the bind operation.
-   *  Possible values: gl.TRANSFORM_FEEDBACK_BUFFER and gl.UNIFORM_BUFFER
-   * @param {GLuint} index - the index of the target.
+   * @param {GLuint} index= - the index of the target.
+   *   - GL.TRANSFORM_FEEDBACK_BUFFER and GL.UNIFORM_BUFFER need an index to affect state
+   * @param {GLuint} offset=0 - the index of the target.
+   *   - GL.UNIFORM_BUFFER: `offset` must be aligned to GL.UNIFORM_BUFFER_OFFSET_ALIGNMENT.
+   * @param {GLuint} size= - the index of the target.
+   *   - GL.UNIFORM_BUFFER: `size` must be a minimum of GL.UNIFORM_BLOCK_SIZE_DATA.
    * @returns {Buffer} - Returns itself for chaining.
    */
-  bindBase({index} = {}) {
-    assertWebGL2Context(this.gl);
-    this.gl.bindBufferBase(this.target, index, this.handle);
-    return this;
-  }
+  bind({target = this.target, index = this.index, offset = 0, size} = {}) {
+    // NOTE: While GL.TRANSFORM_FEEDBACK_BUFFER and GL.UNIFORM_BUFFER could
+    // be used as direct binding points, they will not affect transform feedback or
+    // uniform buffer state. Instead indexed bindings need to be made.
```

Comment:
Let's put the link to the comment here 

https://www.khronos.org/registry/OpenGL-Refpages/es3.0/html/glBindBuffer.xhtml